### DISCUSSION TITLE
Reducer and Orchestration notes

### DISCUSSION THREAD

**Discussion by sakkammadam** [sakkammadam/scratchStuff#9] [Announcements]
@hbaird7877 , @abejr17 

Team - 
 I wanted to give you an update - I have completed the Reducer implementation and added supports to FileProcessor to process its results. This will now produce the final_output folder.

![Screenshot from 2022-10-22 23-07-59](https://user-images.githubusercontent.com/10201590/197376844-874dbbf8-ee67-4a47-bb4c-9fee0b1c0b03.png)

If you step into this folder, you will observe individual output files pertaining to each original file.

![Screenshot from 2022-10-22 23-09-15](https://user-images.githubusercontent.com/10201590/197376875-d7d4b72f-e462-405b-9ad1-fd1672c044e1.png)

If you were to click on each file, it will show the unique words (tokens) and their occurrences within the original file.
![Screenshot from 2022-10-22 23-10-26](https://user-images.githubusercontent.com/10201590/197376925-2f0dad7a-0598-4aa6-b0ac-56d7842345d2.png)

I have also taken the liberty to update main to include mapReduceWorkflow and runOperations functions.

The mapReduceWorkflow function as the name suggests helps orchestrate the entire workflow.

The runOperations function manages the execution within main.

Please take a moment to look at the additions and let me know your thoughts.

So what's left - 

- Could you let me know where you are with boost logging ? I was able to pipe out the logs from streams using the command without boost. I think someone in class was doing the same if I remember - 

  ./runScratch  /home/sakkammadam/Documents/syr/cse687/projects/input_files/shakespeare > logs/run_`date '+%Y%m%d%H%M%S'`.log 

- I am creating a SUCCESS.ind file at the end, maybe we should create this by checking if there is a corresponding output file for an input file.
- Presentation and README.md

Let me know your thoughts. Please let me know if you would like to meet tomorrow. We can discuss branch promotion on the same calls.

Thanks,
Sooraj

**Comment by hbaird7877** [comment]
My Boost logger code is still throwing errors,  I really have little positive to say about Boost.  I spoke to one of my friends from another class who was a C++ developer in the Bay area for several years.  He advised that the use of Boost is generally considered poor practice.  I found another approach to logging using a user-defined class, which I imagine works a lot like what you are doing (I hope soon to be able to get a look at the changes you have made and files you have created.  When I went to Professor Rouche with my concerns about using macOS and other than Windows based tools for this and other projects in this class, he assurred me that you were excellent and would get me through this. Best, Hal
